### PR TITLE
share data_file_id when deserialize put entry

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: ref #6827 

Problem Summary: When deserialize entry from WAL, there may be may duplication in the `data_file_id` field but there is no sharing between them and consume a lot of unnecessary memory.

### What is changed and how it works?
Pass a set as the container for deserialized `data_file_id`, and for every put entry, try to reuse the old `data_file_id` if it already exists.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]
LGTM

File: dbms/src/Storages/Page/V3/WAL/serialize.h [hunk 1]
```diff
@@ -26,12 +26,34 @@ enum WALSerializeVersion : UInt32
     LZ4 = 2,
 };
 
+struct Comparator
+{
+    using is_transparent = void;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/WAL/serialize.h:31]: Wow!

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**